internal/repository/rsl.go

```diff
@@ -90,6 +90,20 @@ func (r *Repository) RecordRSLAnnotation(rslEntryIDs []string, skip bool, messag
 	return rsl.NewAnnotationEntry(rslEntryHashes, skip, message).Commit(r.r, signCommit)
 }
 
+func (r *Repository) InvalidateLatestRSLEntry(ref string) error {
```

Comment:
does this need ref? This is only used when invalidating the absolute latest entry, right? Also, you have to set HEAD to `main` or whatever explicitly, which is likely the issue here...

Reply:
I have fixed this, it turns out I cannnot use `gitinterface.ResetCommit`, I have to use `repo.r.Storer.SetReference`, since reset commit does not actualy change the tip of a ref, it just moves the HEAD pointer, so when I query the latest RSL entry, I get the invalid commit